features/support/commits_finder.rb

```diff
@@ -0,0 +1,39 @@
+# Encapsulates finding Git commits in flexible ways
+#
+# After initializing with the desired attributes for the commit list,
+# run as many #find_* methods as you want to build up your commit list.
+# When done, call #to_a to get the results as a Cucumber compatible array.
+class CommitsFinder
+
+  # commit_attributes - the attributes we want to gather for the commits found
+  #                     The default value is the minimalistic set
+  #                     for checking if commits exist at all
+  #
+  # Examples
+  #
+  #   CommitsFinder.new [:branch, :location, :message, :file_name, :file_content, :author]
+  def initialize commit_attributes = [:message]
+    @commits = CommitList.new commit_attributes
```

Comment:
I'm still against this having internal state. I like the idea that the find methods return a CommitList but multiple calls to `find_*` should not effect each other. If you want to avoid `find_commit_in_current_repo` from having to concatenate multiple commit lists, that can still be done
```ruby
  def initialize commit_attributes = [:message]
    @commit_attributes = commit_attributes
  end


  def find_commits_in_branch branch_name, commit_list = nil
    commit_list ||= new CommitList @commit_attributes
    array_output_of("git log #{branch_name} --format='%h|%s|%ae' --topo-order --reverse").each do |commit|
      sha, message, author = commit.split('|')
      next if message == 'Initial commit'
      commit_list.add sha: sha, message: message, branch_name: branch_name, author: author
    end
    commit_list
  end


  def find_commits_in_current_repo commit_list = nil
    commit_list ||= new CommitList @commit_attributes
    existing_branches.each do |branch_name|
      find_commits_in_branch branch_name, commit_list
    end
    commit_list
  end

end
```